src/Features/CSharp/Portable/SplitOrMergeIfStatements/CSharpIfLikeStatementGenerator.cs

```diff
@@ -0,0 +1,151 @@
+        }
+
+        public bool IsElseIfClause(SyntaxNode node, out SyntaxNode parentIfLikeStatement)
+        {
+            if (node is IfStatementSyntax && node.Parent is ElseClauseSyntax)
+            {
+                parentIfLikeStatement = (IfStatementSyntax)node.Parent.Parent;
+                return true;
+            }
+
+            parentIfLikeStatement = null;
+            return false;
+        }
+
+        public SyntaxNode GetCondition(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.Condition;
+        }
+
+        public SyntaxNode GetRootIfStatement(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+
+            while (ifStatement.Parent is ElseClauseSyntax elseClause)
+            {
+                ifStatement = (IfStatementSyntax)elseClause.Parent;
+            }
+
+            return ifStatement;
+        }
+
+        public ImmutableArray<SyntaxNode> GetElseLikeClauses(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+
+            var builder = ImmutableArray.CreateBuilder<SyntaxNode>();
+
+            while (ifStatement.Else?.Statement is IfStatementSyntax elseIfStatement)
+            {
+                builder.Add(elseIfStatement);
+                ifStatement = elseIfStatement;
+            }
+
+            if (ifStatement.Else != null)
+            {
+                builder.Add(ifStatement.Else);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public SyntaxNode WithCondition(SyntaxNode ifLikeStatement, SyntaxNode condition)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithCondition((ExpressionSyntax)condition);
+        }
+
+        public SyntaxNode WithStatement(SyntaxNode ifLikeStatement, SyntaxNode statement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithStatement(SyntaxFactory.Block((StatementSyntax)statement));
+        }
+
+        public SyntaxNode WithStatementsOf(SyntaxNode ifLikeStatement, SyntaxNode otherIfLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            var otherIfStatement = (IfStatementSyntax)otherIfLikeStatement;
+            return ifStatement.WithStatement(otherIfStatement.Statement);
+        }
+
+        public SyntaxNode WithElseLikeClausesOf(SyntaxNode ifStatement, SyntaxNode otherIfStatement)
+        {
+            return ((IfStatementSyntax)ifStatement).WithElse(((IfStatementSyntax)otherIfStatement).Else);
+        }
+
+        public SyntaxNode ToIfStatement(SyntaxNode ifLikeStatement)
+            => ifLikeStatement;
+
+        public SyntaxNode ToElseIfClause(SyntaxNode ifLikeStatement)
+            => ((IfStatementSyntax)ifLikeStatement).WithElse(null);
```

Comment:
this method body doesn't seem to match waht i would have expected of the name.  why is "ToElseIfClause something that removes an 'else' in C#?

Reply:
We're considering an else-if clause in the VB sense of ElseIf, meaning it is independent of any other else-ifs and else clauses, so if you insert it as an else-if clause somewhere, it's "else" will be ignored anyway.

Reply:
The doc comment in the interface mentions that the else would be dropped.